project/src/main/java/com/google/sps/servlets/LoadExploreServlet.java

```diff
@@ -26,48 +26,62 @@
+import java.util.Set;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import com.google.sps.data.User;
 import com.google.appengine.api.users.UserService;
 import com.google.appengine.api.users.UserServiceFactory;
-
+ 
 /** Servlet that loads pending events*/
 @WebServlet("/load-explore")
 public class LoadExploreServlet extends HttpServlet {
-
+ 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Query query = new Query("ApprovedEvent");
     query.addSort("dateTimestamp", SortDirection.DESCENDING);
-
+ 
     UserService userService = UserServiceFactory.getUserService();
     if (userService.isUserLoggedIn()) {
-
+      String email = userService.getCurrentUser().getEmail().toLowerCase();
+ 
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
       PreparedQuery results = datastore.prepare(query);
       List<Entity> resultsList = results.asList(FetchOptions.Builder.withDefaults());
       List<Event> exploreEvents = new ArrayList<>();
-
+ 
       for (int i = 0; i < resultsList.size(); i++) {
-
+ 
         Entity entity = resultsList.get(i);
-        long id = entity.getKey().getId();
-        String name = (String) entity.getProperty("name");
-        String date = (String) entity.getProperty("date");
-        String description = (String) entity.getProperty("description");
-        String type = (String) entity.getProperty("attendance");
-        String attendance = (String) entity.getProperty("type");
-        long timestamp = (long) entity.getProperty("timestamp");
-
-        Event event = new Event(id, name, date, description, type, attendance, timestamp, false, false);
-        exploreEvents.add(event);
+        // Load the event only if this user's email matches the email of an attendee
+        @SuppressWarnings("unchecked") // Cast can't verify generic type.
+        Collection<String> attendees = (Collection<String>) entity.getProperty("attendees");
+        if (!attendees.isEmpty()) {
+          if (!attendees.contains(email)) {
+            long id = entity.getKey().getId();
+            String name = (String) entity.getProperty("name");
+            String location = (String) entity.getProperty("location");
+            String date = (String) entity.getProperty("date");
+            String time = (String) entity.getProperty("time");
+            String description = (String) entity.getProperty("description");
+            String type = (String) entity.getProperty("attendance");
+            String attendance = (String) entity.getProperty("type");
+            long timestamp = (long) entity.getProperty("timestamp");
+ 
+            Event event = new Event(id, name, location, date, time, description, type, attendance, timestamp, entity.getProperty("email").equals(email), "ApprovedEvent");
```

Comment:
extract `entity.getProperty("email").equals(email)` to a var for easy understanding.